Bottom: a0c673156cb90c6eb58a8b7bee0cc85c00411a74
Top:    2bea3b810f0516eef0624fe1983f9db0b7e34176
Author: Alexey Lyashkov <shadow@Alexeys-MacBook-Pro.local>
Date:   2016-02-02 16:47:27 +0300

Refresh of ctl-verbs.patch

---

diff --git a/cli.c b/cli.c
index 2732030..aa5a5a0 100644
--- a/cli.c
+++ b/cli.c
@@ -36,16 +36,16 @@ cli_init(void)
 		printk("error connect \n");
 		goto exit;
 	}
-	event = ib_socket_poll(sock);
-	printk("Event hit %lx\n", event);
-	if (event & POLLOUT) {
-		err = ib_socket_write(sock, srv_addr, strlen(srv_addr));
-		printk("status send %d\n", err);
+	while ( 1 ) {
+		event = ib_socket_poll(sock);
+		printk("Event hit %lx\n", event);
+		if (event & POLLERR)
+			break;
+		if (event & POLLOUT) {
+			err = ib_socket_write(sock, srv_addr, strlen(srv_addr));
+			printk("sends status %d\n", err);
+		}
 	}
-	/* next event when transfer is done */
-	printk("Event hit %lx\n", event);
-	event = ib_socket_poll(sock);
-
 
 	ib_socket_disconnect(sock);
 exit:
